### PR TITLE
feat: routing form fields as variables in route

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Allows using a field identifier as a variable for event type redirects. 

Loom: https://www.loom.com/share/2bd8b552028740fc9c4663e0cff08eb9

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- See loom 

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx [hunk 1]
```diff
@@ -101,7 +102,25 @@ function RoutingForm({ form, profile, ...restProps }: Props) {
...
+            const identifier = getFieldIdentifier(fields.find((field) => field.id === key));
+            if (identifier === variable) {
+              eventTypeUrl = eventTypeUrl.replace(
+                `{${variable}}`,
+                slugify(response[key].value.toString() || "")
+              );
+            }
+          }
+        });
+        await router.push(`/${eventTypeUrl}?${allURLSearchParams}`);
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx:123]: We happened  to change the same piece of code for 2 different reasons  😅

Comment (CarinaWolli) [reply packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx:123]: yeah I saw that too 😅 we can merge yours first and I'll handle the conflicts

Comment (hariombalhara) [reply packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx:123]: Mine is not that urgent. Happy to merge it first. Could you extract out this logic in a separate function

**Comment by hariombalhara** [comment]
Looks like the perfect solution. 

We can add some improvements later  like 
- User isn't able to add variables that don't exist in the Form
- Whenever user changes an identifier and it's used in a route, we update it there as well.

File: packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx [hunk 2]
```diff
@@ -128,6 +130,15 @@ const Route = ({
     });
   });
 
+  const eventTypePrefix =
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx:133]:
```suggestion
 // /team/{TEAM_SLUG}/{EVENT_SLUG} -> /team/{TEAM_SLUG}
  const eventTypePrefix =
```

A comment would be helpful here.

File: packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx [hunk 3]
```diff
@@ -101,7 +102,25 @@ function RoutingForm({ form, profile, ...restProps }: Props) {
...
+
+        const variables: string[] =
+          decidedAction.value.match(regex)?.map((match: string) => match.slice(1, -1)) || [];
+
+        let eventTypeUrl = decidedAction.value;
+
+        variables.forEach((variable) => {
+          for (const key in response) {
+            const identifier = getFieldIdentifier(fields.find((field) => field.id === key));
+            if (identifier === variable) {
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx:115]: Could be better to do a case-insensitive match?

**Review by hariombalhara — DISMISSED** [review]
LGTM !!

File: packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx (hunk 1 above)
Comment (CarinaWolli) [reply packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx]: extracted it into a separate function, @hariombalhara can you re-review?

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**